docs/src/design-system/building-blocks/button/ButtonStatesExample.tsx

```diff
@@ -1,107 +1,127 @@
-import { Box } from '@mui/material';
 import {
+  Box,
   Table,
   TableBody,
   TableCell,
   tableCellClasses,
   TableContainer,
   TableHead,
   TableRow,
+  Theme,
   ThemeProvider,
   useTheme,
 } from '@mui/material';
 import { Button, Typography } from '@ui-kit-2022/components';
-import { dark } from '@ui-kit-2022/theme';
 import React from 'react';
 
-interface Props {
+export interface Props {
   variantType?: 'PRIMARY' | 'SECONDARY' | 'TERTIARY';
+  theme: Theme;
 }
-const DarkBgState: React.FC<Props> = ({ variantType }) => {
-  const { palette } = useTheme();
+
+const ButtonStatesExample: React.FC<Props> = ({ variantType, theme = useTheme() }) => {
```

Comment:
Hmm... I wouldn't put a React hook in component props like this as a default value, it isn't common practice.

I would suggest instead maybe provide `overrideTheme` as an optional prop and begin the component with `const globalTheme = useTheme();` as usual and then you can derive `const theme = overrideTheme ?? globalTheme;` should get you the override theme if its defined, otherwise the global theme.